feugy/melodie
# feat: improves usage on mobile and updates dependencies

### What's in there?

When using Mélodie as my music player on an Android phone, a flaky connectivity can quickly become an annoying issue.
I ran multiple times into a situation where I had to get my phone, unlock it, and refresh the browser to get to the next track.

Are included here:

- [x] feat(ui): automatically retries downloading played track on network error
- [x] feat(ui): disable player buttons while downloading played track
- [x] feat(ui): sends console errors and warning to the backend
- [x] fix(ui): can not add many tracks to playlist (Paul Dinletir tracks)
- [x] fix(ui): some playlists are crashing the app
- [x] fix(ui): Android does not supports notifications
- [x] fix(ui): pre-fetching next track does not work
- [x] fix(core): some track list contain empty tracks
- [x] refactor(ui): applies Tabulous improvements on DropDown menu
- [x] refactor(ui): uses @testing-library user-event rather than fireEvent
- [x] refactor(core): improves error serialization in logs
- [x] chore(ui): updates to latest caniuse version
- [x] chore(site): updates copyright date
- [x] chore: updates all dependencies but tailwindcss and postcss (issues with jest and storybook)

### How to test?

1. start melodie locally
2. enables broadcast
3. connect to the app through your mobile
4. play a song on mobile
5. disable broadcast on desktop
6. go to next song: it should play thanks to caching
7. enable broadcast on desktop
8. go to next song: everything works smoothly.
